Data-Science/Kubeflow/Lightweight-Python-Functions/lightweight_python_functions_v2.ipynb

```diff
@@ -0,0 +1,204 @@
+    "def train(\n",
+    "    # Use InputPath to get a locally accessible path for the input artifact\n",
+    "    # of type `Dataset`.\n",
+    "    dataset_one_path: InputPath('Dataset'),\n",
+    "    # Use Input[T] to get a metadata-rich handle to the input artifact\n",
+    "    # of type `Dataset`.\n",
+    "    dataset_two: Input[Dataset],\n",
+    "    # An input parameter of type string.\n",
+    "    message: str,\n",
+    "    # Use Output[T] to get a metadata-rich handle to the output artifact\n",
+    "    # of type `Dataset`.\n",
+    "    model: Output[Model],\n",
+    "    # An input parameter of type bool.\n",
+    "    input_bool: bool,\n",
+    "    # An input parameter of type dict.\n",
+    "    input_dict: Dict[str, int],\n",
+    "    # An input parameter of type List[str].\n",
+    "    input_list: List[str],\n",
+    "    # An input parameter of type int with a default value.\n",
+    "    num_steps: int = 100,\n",
+    "):\n",
+    "    \"\"\"Dummy Training step.\"\"\"\n",
+    "    with open(dataset_one_path, 'r') as input_file:\n",
+    "        dataset_one_contents = input_file.read()\n",
+    "\n",
+    "    with open(dataset_two.path, 'r') as input_file:\n",
+    "        dataset_two_contents = input_file.read()\n",
+    "\n",
+    "    line = (f'dataset_one_contents: {dataset_one_contents} || '\n",
+    "            f'dataset_two_contents: {dataset_two_contents} || '\n",
+    "            f'message: {message} || '\n",
+    "            f'input_bool: {input_bool}, type {type(input_bool)} || '\n",
+    "            f'input_dict: {input_dict}, type {type(input_dict)} || '\n",
+    "            f'input_list: {input_list}, type {type(input_list)} \\n')\n",
+    "\n",
+    "    with open(model.path, 'w') as output_file:\n",
+    "        for i in range(num_steps):\n",
+    "            output_file.write('Step {}\\n{}\\n=====\\n'.format(i, line))\n",
+    "\n",
+    "    # model is an instance of Model artifact, which has a .metadata dictionary\n",
+    "    # to store arbitrary metadata for the output artifact.\n",
+    "    model.metadata['accuracy'] = 0.9"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ee649616-7cb9-464d-a4e5-987baec001d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.pipeline(name='lightweight-python-functions')\n",
+    "def pipeline(message: str = 'message'):\n",
+    "    preprocess_task = preprocess(message=message)\n",
+    "    train_task = train(\n",
+    "        dataset_one_path=preprocess_task.outputs['output_dataset_one'],\n",
+    "        dataset_two=preprocess_task.outputs['output_dataset_two_path'],\n",
+    "        message=preprocess_task.outputs['output_parameter_path'],\n",
+    "        input_bool=preprocess_task.outputs['output_bool_parameter_path'],\n",
+    "        input_dict=preprocess_task.outputs['output_dict_parameter_path'],\n",
+    "        input_list=preprocess_task.outputs['output_list_parameter_path'],\n",
```

Comment:
Wouldn't it be better to rename in/out parameters for better consistency like `output_parameter_path` -> `output_string_parameter_path` or `message` -> `input_string`?